### PR TITLE
[dead-code] Remove few unused classes

### PR DESCRIPTION
**Pull request by TomasVotruba**
The best code to upgrade, is no code :slightly_smiling_face: 

Before jumping to filling last few types, it would be effective to remove classes that are never used.
These classes are checked by class-leak tool and spotted as unused. Roughly ~1100, but I have to go through false positives.

Would you be interested in such a removal?

---

## Try it yourself

```bash
bin/class-leak check app plugins --skip-suffix="EmojiMap"
```

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Let's kick of with first 5, to see how CI accepts it and if there is an interest :+1:

**Comment by TomasVotruba** [comment]
>  will increase coverage by 0.02%.

Nice side effect :smile:

**Comment by TomasVotruba** [comment]
Ready for review :+1:  

If this gets accepted, I'll send few more removing classes in parallel, so we have fast feedback.

File: app/bundles/IntegrationsBundle/Auth/Provider/Oauth2ThreeLegged/Credentials/StateInterface.php [hunk 1]
```diff
@@ -1,10 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Mautic\IntegrationsBundle\Auth\Provider\Oauth2ThreeLegged\Credentials;
-
-interface StateInterface
```
Comment (escopecz) [inline app/bundles/IntegrationsBundle/Auth/Provider/Oauth2ThreeLegged/Credentials/StateInterface.php:7]: It is tempting! But for example in this case I found that we have 3 plugins using this interface just at Acquia. There may be more from other vendors. Although I love removing code, I think we shouldn't remove classes. We should not introduce more BC breaks in the RC versions. We can mark them as deprecated and see how many people will scream. If none, we can remove them in Mautic 6.

**Comment by escopecz** [comment]
I run the command `bin/class-leak check app plugins --skip-suffix="EmojiMap"` and I see it curated a list of `Found 1188 unused classes.`

I expanded the command to:

```
bin/class-leak check app plugins --skip-suffix="EmojiMap" --skip-suffix="Controller" --skip-suffix="Test" --skip-suffix="Middleware" --skip-suffix="Extension" --skip-suffix="Bundle" --skip-suffix="Subscriber" --skip-suffix="Permissions" --skip-suffix="Command"
```
Which returns 63 classes:
```
Classes with a parent/interface - possibly used by type
=======================================================

 * AppCache
app/AppCache.php

 * Mautic\ApiBundle\Entity\oAuth2\RefreshToken
app/bundles/ApiBundle/Entity/oAuth2/RefreshToken.php

 * Mautic\CampaignBundle\Executioner\Scheduler\Exception\ExecutionProhibitedException
app/bundles/CampaignBundle/Executioner/Scheduler/Exception/ExecutionProhibitedException.php

 * Mautic\CampaignBundle\Executioner\Scheduler\Exception\NotTimeYetException
app/bundles/CampaignBundle/Executioner/Scheduler/Exception/NotTimeYetException.php

 * Mautic\CampaignBundle\Form\Type\CampaignEventLeadChangeType
app/bundles/CampaignBundle/Form/Type/CampaignEventLeadChangeType.php

 * Mautic\CoreBundle\Form\DataTransformer\DatetimeToStringTransformer
app/bundles/CoreBundle/Form/DataTransformer/DatetimeToStringTransformer.php

 * Mautic\CoreBundle\Form\DataTransformer\EmojiToHtmlTransformer
app/bundles/CoreBundle/Form/DataTransformer/EmojiToHtmlTransformer.php

 * Mautic\CoreBundle\Form\DataTransformer\NullToEmptyTransformer
app/bundles/CoreBundle/Form/DataTransformer/NullToEmptyTransformer.php

 * Mautic\CoreBundle\Session\Storage\Handler\RedisSentinelSessionHandler
app/bundles/CoreBundle/Session/Storage/Handler/RedisSentinelSessionHandler.php

 * Mautic\CoreBundle\Test\MauticSqliteTestCase
app/bundles/CoreBundle/Test/MauticSqliteTestCase.php

 * Mautic\EmailBundle\Form\Type\AbTestPropertiesType
app/bundles/EmailBundle/Form/Type/AbTestPropertiesType.php

 * Mautic\EmailBundle\Validator\DsnValidator
app/bundles/EmailBundle/Validator/DsnValidator.php

 * Mautic\FormBundle\Tests\FormTestAbstract
app/bundles/FormBundle/Tests/FormTestAbstract.php

 * Mautic\IntegrationsBundle\Auth\Provider\Oauth1aThreeLegged\CredentialsInterface
app/bundles/IntegrationsBundle/Auth/Provider/Oauth1aThreeLegged/CredentialsInterface.php

 * Mautic\IntegrationsBundle\Auth\Provider\Oauth2ThreeLegged\Credentials\AccessTokenInterface
app/bundles/IntegrationsBundle/Auth/Provider/Oauth2ThreeLegged/Credentials/AccessTokenInterface.php

 * Mautic\IntegrationsBundle\Auth\Provider\Oauth2ThreeLegged\Credentials\RefreshTokenInterface
app/bundles/IntegrationsBundle/Auth/Provider/Oauth2ThreeLegged/Credentials/RefreshTokenInterface.php

 * Mautic\IntegrationsBundle\Exception\IntegrationMappingException
app/bundles/IntegrationsBundle/Exception/IntegrationMappingException.php

 * Mautic\IntegrationsBundle\Exception\InvalidProviderException
app/bundles/IntegrationsBundle/Exception/InvalidProviderException.php

 * Mautic\IntegrationsBundle\Form\Type\ActivityListType
app/bundles/IntegrationsBundle/Form/Type/ActivityListType.php

 * Mautic\LeadBundle\Form\Validator\Constraints\UniqueCustomFieldValidator
app/bundles/LeadBundle/Form/Validator/Constraints/UniqueCustomFieldValidator.php

 * Mautic\MarketplaceBundle\Exception\InstallException
app/bundles/MarketplaceBundle/Exception/InstallException.php

 * Mautic\MessengerBundle\DependencyInjection\EnvProcessor\MessengerNullableEnvVarProcessor
app/bundles/MessengerBundle/DependencyInjection/EnvProcessor/MessengerNullableEnvVarProcessor.php

 * Mautic\MessengerBundle\MessageHandler\TestHandler
app/bundles/MessengerBundle/MessageHandler/TestHandler.php

 * Mautic\MessengerBundle\Transport\NullTransportFactory
app/bundles/MessengerBundle/Transport/NullTransportFactory.php

 * Mautic\MessengerBundle\Validator\DsnValidator
app/bundles/MessengerBundle/Validator/DsnValidator.php

 * Mautic\NotificationBundle\Event\NotificationClickEvent
app/bundles/NotificationBundle/Event/NotificationClickEvent.php

 * Mautic\NotificationBundle\Event\SendingNotificationEvent
app/bundles/NotificationBundle/Event/SendingNotificationEvent.php

 * Mautic\NotificationBundle\Form\Type\ConfigType
app/bundles/NotificationBundle/Form/Type/ConfigType.php

 * Mautic\PageBundle\Form\Type\PagePublishDatesType
app/bundles/PageBundle/Form/Type/PagePublishDatesType.php

 * Mautic\CodingStandards\PhpCSFixer\NoTablePrefixDefinitionInTestsFixer
app/codingstandards/PhpCSFixer/NoTablePrefixDefinitionInTestsFixer.php

 * MauticPlugin\MauticCrmBundle\Api\VtigerApi
plugins/MauticCrmBundle/Api/VtigerApi.php

 * MauticPlugin\MauticEmailMarketingBundle\Api\ConstantContactApi
plugins/MauticEmailMarketingBundle/Api/ConstantContactApi.php

 * MauticPlugin\MauticEmailMarketingBundle\Api\IcontactApi
plugins/MauticEmailMarketingBundle/Api/IcontactApi.php

 * MauticPlugin\MauticEmailMarketingBundle\Api\MailchimpApi
plugins/MauticEmailMarketingBundle/Api/MailchimpApi.php

 * MauticPlugin\MauticFullContactBundle\Services\FullContact_API
plugins/MauticFullContactBundle/Services/FullContact_API.php

 * MauticPlugin\MauticSocialBundle\Form\Type\TwitterCustomType
plugins/MauticSocialBundle/Form/Type/TwitterCustomType.php


Classes without any parent/interface - easier to remove
=======================================================

 * Mautic\CoreBundle\Doctrine\TranslationMigrationTrait
app/bundles/CoreBundle/Doctrine/TranslationMigrationTrait.php

 * Mautic\CoreBundle\Doctrine\VariantMigrationTrait
app/bundles/CoreBundle/Doctrine/VariantMigrationTrait.php

 * Mautic\CoreBundle\Entity\Cache
app/bundles/CoreBundle/Entity/Cache.php

 * Mautic\CoreBundle\Entity\UpsertTrait
app/bundles/CoreBundle/Entity/UpsertTrait.php

 * Mautic\CoreBundle\Entity\UuidInterface
app/bundles/CoreBundle/Entity/UuidInterface.php

 * Mautic\CoreBundle\Entity\UuidTrait
app/bundles/CoreBundle/Entity/UuidTrait.php

 * Mautic\IntegrationsBundle\Form\Type\NotBlankIfPublishedConstraintTrait
app/bundles/IntegrationsBundle/Form/Type/NotBlankIfPublishedConstraintTrait.php
```
I removed migrations from the list above so it's shorter. I'm not sure how to exclude those as they don't have common suffix. I'll leave the list here so anyone (probably me) could take it and validate and deprecate those classes once we release Mautic 5.

Would you mind if I close this PR for now?

**Comment by escopecz** [comment]
I went through that list and created https://github.com/mautic/mautic/pull/13066 with deprecations. So closing this PR. Thanks @TomasVotruba for pointing those out!

**Comment by TomasVotruba** [comment]
Thanks for checking. I didn't know this is being extended outside this repository. I'm find with closing :+1:

**Pull request closed without merging**